src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -544,6 +544,60 @@ impl<'a, 'tcx> LinkCollector<'a, 'tcx> {
             })
     }
 
+    /// Resolve a type alias for the purposes of intra-doc links.
+    fn resolve_type_alias(&self, ty_id: DefId) -> Option<Res> {
+        use PrimitiveType::*;
+        let ty = self.cx.tcx.type_of(ty_id);
+        Some(match ty.kind() {
```

Comment:
Are you sure no one has needed to do this before? It seems really strange we'd have to rewrite it here, I'd expect there to be a way to do it in rustc_middle, at least to get the DefId if it exists.

Reply:
If I'm right in https://github.com/rust-lang/rust/pull/86334#discussion_r653912376, `inherent_impls` won't return anything. I think `rustdoc` is able to display the implementations for type aliases because it does its own visit of the HIR to collect them. Maybe there's a way to use that here, but when I tried the cache was still empty, I'm assuming it gets populated later in the process.

Reply:
I was surprised too, but I couldn't find anything. Maybe I should ask on Zulip.
But for our purposes here, most of the churn is converting the primitive types to `Res::Primitive`, so I'm not really sure a function in `rustc_middle` would be too useful